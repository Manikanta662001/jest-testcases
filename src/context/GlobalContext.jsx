import React, { createContext, useContext, useEffect, useState } from "react";

export const GlobalContext = createContext(null);

function AuthContext({ children }) {
  const [currentUser, setCurrentUser] = useState({});
  const [isLoggedIn, setIsLoggedIn] = useState(false);
  const alldata = { currentUser, setCurrentUser, isLoggedIn, setIsLoggedIn };
  return (
    <>
      <GlobalContext.Provider value={alldata}>
        {children}
      </GlobalContext.Provider>
    </>
  );
}
export const useGlobalContext = () => useContext(GlobalContext);
export default AuthContext;
