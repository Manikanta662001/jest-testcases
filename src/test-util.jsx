// src/test-util.jsx
import React from "react";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
// Import additional providers if needed, e.g., AuthProvider

const customRender = (ui, { route = "/", ...options } = {}) => {
  window.history.pushState({}, "Test page", route);

  const Wrapper = ({ children }) => (
    <BrowserRouter>
      {/* Include other providers here if necessary */}
      {children}
    </BrowserRouter>
  );

  return render(ui, { wrapper: Wrapper, ...options });
};

export * from "@testing-library/react";
export { customRender as render };
