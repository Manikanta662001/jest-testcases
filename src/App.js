import React from "react";
import { useGlobalContext } from "./context/GlobalContext";
import { Navigate, Route, Routes } from "react-router-dom";
import Register from "./pages/Register";
import Login from "./pages/Login";
import Dashboard from "./pages/Dashboard";
import ErrorPage from "./pages/ErrorPage";

function App() {
  const userContext = useGlobalContext();
  const { isLoggedIn } = userContext;
  return (
    <div>
      <Routes>
        <Route path="/register" element={<Register />} />
        <Route
          path="/"
          element={isLoggedIn ? <Navigate to={"/dashboard"} /> : <Login />}
        />
        <Route
          path="/dashboard"
          element={isLoggedIn ? <Dashboard /> : <Navigate to={"/"} />}
        />
        <Route path="*" element={<ErrorPage />} />
      </Routes>
    </div>
  );
}

export default App;
