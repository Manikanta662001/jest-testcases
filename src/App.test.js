import { screen } from "@testing-library/react";
import App from "./App";
import { render } from "./test-util";
import { GlobalContext } from "./context/GlobalContext";
import React from "react";

const contextValues = {
  currentUser: {},
  setCurrentUser: jest.fn(),
  isLoggedIn: false,
  setIsLoggedIn: jest.fn(),
};
describe("App Component", () => {
  test("should Match snapshot", () => {
    const { asFragment } = render(
      <GlobalContext.Provider value={contextValues}>
        <App />
      </GlobalContext.Provider>
    );
    expect(asFragment(<App />)).toMatchSnapshot();
  });
  test("should show login page ", () => {
    render(
      <GlobalContext.Provider value={contextValues}>
        <App />
      </GlobalContext.Provider>,
      { route: "/" }
    );
    expect(screen.getByText("Login Page")).toBeInTheDocument();
  });
  test("should show register page ", () => {
    render(
      <GlobalContext.Provider value={contextValues}>
        <App />
      </GlobalContext.Provider>,
      { route: "/register" }
    );
    expect(screen.getByText("Register Page")).toBeInTheDocument();
  });
  test("should show Dashboard page ", () => {
    render(
      <GlobalContext.Provider value={{ ...contextValues, isLoggedIn: true }}>
        <App />
      </GlobalContext.Provider>,
      { route: "/dashboard" }
    );
    expect(screen.getByText("Dashboard Page")).toBeInTheDocument();
  });
  test("should show Dashboard page if we loggedin and if we goto login page", () => {
    render(
      <GlobalContext.Provider value={{ ...contextValues, isLoggedIn: true }}>
        <App />
      </GlobalContext.Provider>,
      { route: "/" }
    );
    expect(screen.getByText("Dashboard Page")).toBeInTheDocument();
  });
});
