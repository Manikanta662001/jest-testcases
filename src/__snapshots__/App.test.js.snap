// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should Match snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      Login Page
    </div>
  </div>
</DocumentFragment>
`;
